src/js/common/schemaform/SelectiveArrayField.jsx

```diff
@@ -0,0 +1,203 @@
+import _ from 'lodash/fp';
+import classNames from 'classnames';
+import Scroll from 'react-scroll';
+
+
+import {
+  toIdSchema,
+  getDefaultFormState,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+
+class SelectiveArrayField extends React.Component {
+  // This fills in an empty item in the array if it has minItems set
+  // so that schema validation runs against the fields in the first item
+  // in the array. This shouldn't be necessary, but there's a fix in rjsf
+  // that has not been released yet
+  componentDidMount = () => {
+    const { schema, formData = [], registry } = this.props;
+    if (schema.minItems > 0 && formData.length === 0) {
+      this.props.onChange(Array(schema.minItems).fill(
+        getDefaultFormState(schema.additionalItems, undefined, registry.definitions)
+      ));
+    }
+  }
+
+  shouldComponentUpdate = (nextProps, nextState) => {
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
+  }
+
+  onItemChange = (indexToChange, value) => {
+    const newItems = _.set(indexToChange, value, this.props.formData || []);
+    this.props.onChange(newItems);
+  }
+
+  getItemSchema = (index, selectedFields) => {
+    const schema = this.props.schema;
+    const itemSchema = schema.items.length > index
+      ? schema.items[index]
+      : schema.additionalItems;
+
+    const properties = _.pick(selectedFields, itemSchema.properties);
+    const required = _.intersection(selectedFields, itemSchema.required);
+
+    return _.assign(itemSchema, {
+      properties,
+      required
+    });
+  }
+
+  scrollToTop = () => {
+    setTimeout(() => {
+      scroller.scrollTo(`topOfTable_${this.props.idSchema.$id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: -60
+      });
+    }, 100);
+  }
+
+  scrollToRow = (id) => {
+    setTimeout(() => {
+      scroller.scrollTo(`table_${id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: 0
+      });
+    }, 100);
+  }
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      disabled,
+      readonly,
+      registry,
+      formContext,
+      onBlur,
+      schema
+    } = this.props;
+    const definitions = registry.definitions;
+    const { TitleField, SchemaField } = registry.fields;
+
+    const title = uiSchema['ui:title'] || schema.title;
+    const hideTitle = !!_.get(['ui:options', 'hideTitle'], uiSchema);
+    const description = uiSchema['ui:description'];
+    const textDescription = typeof description === 'string' ? description : null;
+    const DescriptionField = typeof description === 'function'
+      ? uiSchema['ui:description']
+      : null;
+    const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
+    const selectedFields = uiSchema['ui:selectedFields'] || Object.keys(schema.properties);
```

Comment:
I think the disconnect is that I'm thinking of the child definitions for the two pages as largely separate, but you built the child definition as one thing. So you're coming from the angle of needing to split the schema and uiSchema apart, but I have been assuming it's built as two separate definitions already. 

So there are two approaches: build two separate definitions, and make the SelectiveArrayField component essentially the normal ArrayField component but stripped of any way to modify the number of rows. The second approach is the direction you're going, which is having one overall child definition, and have SelectiveArrayField show particular fields based on uiSchema configuration info. Me pushing for removing some of what you have in SelectiveArrayField to select fields puts us somewhere in the middle of the two approaches, which is probably a bad place. We should just pick one or the other.

I have been assuming the first approach because I don't see much (or any) overlap in the uiSchemas between the two pages, and it seems straightforward to use `pick` and `omit` to pull the fields out of the `child` schema definition (similar to how we pass lists of fields to other common definition files) for each page. So building two separate definitions seemed clearer to me.

I think this pattern is probably limited to HCA, and the changes to `uiSchema` are probably only going to impact this particular array component. So ultimately, I think either approach is fine. I think you should pick whichever one seems cleaner to you.